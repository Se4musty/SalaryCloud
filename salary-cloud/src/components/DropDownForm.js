import React from 'react';
import Button from 'react-bootstrap/Button';
import InputGroup from 'react-bootstrap/InputGroup';
import FormControl from 'react-bootstrap/FormControl';
import { Link } from 'react-router-dom';
import api from '../api/api';

class DropDownForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: '',
      passwordConfirmation: ''
    };
  }

  handleChangeInputEmail = async (event) => {
    const emailInput = event.target.value;
    this.setState({ email: emailInput });
  }

  handleChangeInputPassword = async (event) => {
    const passwordInput = event.target.value;
    this.setState({ password: passwordInput });
  }

  handleChangeInputPasswordConfirmation = async (event) => {
    const passwordConfirmationInput = event.target.value;
    this.setState({ passwordConfirmation: passwordConfirmationInput });
  }

  handleCreateUser = async (event) => {
    const {email, password, passwordConfirmation} = this.state;
    if (password !== passwordConfirmation) {
      window.alert("Passwords don't match!");
    } else {
      await api.createUser({email, hash: password}).then(response => {
        window.alert("User created!");
        this.setState({
          email: '',
          password: '',
          passwordConfirmation: ''
        });
      });
      // TODO: REDIRECT TO ProfileSettings Page
    }

  }

  render () {
    const {email, password, passwordConfirmation} = this.state;

    return(
      <div className="mx-2">
          <InputGroup className="mb-3">
            <FormControl
              type="email"
              placeholder="Email"
              aria-label="Email"
              aria-describedby="email-input"
              value={email}
              onChange={this.handleChangeInputEmail}
            />
          </InputGroup>

          <InputGroup className="mb-3">
            <FormControl
              type="password"
              placeholder="Password"
              aria-label="Password"
              aria-describedby="password-input"
              value={password}
              onChange={this.handleChangeInputPassword}
            />
          </InputGroup>

          <InputGroup className="mb-3">
            <FormControl
              type="password"
              placeholder="Confirm Password"
              aria-label="Confirm Password"
              aria-describedby="confirmation-input"
              value={passwordConfirmation}
              onChange={this.handleChangeInputPasswordConfirmation}
            />
          </InputGroup>

            <Button variant="primary" onClick={this.handleCreateUser}>
              Submit
            </Button>{' '}
        </div>
    );

  }
}

export default DropDownForm;
