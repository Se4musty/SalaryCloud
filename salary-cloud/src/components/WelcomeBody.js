import React from 'react';
import './WelcomeBody.css';

function WelcomeBody() {
  return (
    <div className="WelcomeBody">
      <h1>Welcome to SalaryCloud</h1>
      <button>Create Account</button>
    </div>
  );
}

export default WelcomeBody;