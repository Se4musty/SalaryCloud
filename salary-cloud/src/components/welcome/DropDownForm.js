import React from 'react';
import Button from 'react-bootstrap/Button';
import InputGroup from 'react-bootstrap/InputGroup';
import FormControl from 'react-bootstrap/FormControl';
import { Redirect } from 'react-router-dom';
import usersApi from '../../api/users-api';
import anonUsersApi from '../../api/anon-users-api';

class DropDownForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      toProfileSettings: false,
      email: '',
      password: '',
      passwordConfirmation: ''
    };
  }

  handleChangeInputEmail = async (event) => {
    const emailInput = event.target.value;
    this.setState({ email: emailInput });
  }

  handleChangeInputPassword = async (event) => {
    const passwordInput = event.target.value;
    this.setState({ password: passwordInput });
  }

  handleChangeInputPasswordConfirmation = async (event) => {
    const passwordConfirmationInput = event.target.value;
    this.setState({ passwordConfirmation: passwordConfirmationInput });
  }

  handleCreateUser = async (event) => {
    const {email, password, passwordConfirmation} = this.state;
    if (password !== passwordConfirmation) {
      window.alert("Passwords don't match!");
    } else {
      window.alert("Please wait while profile is being created...");
      await usersApi.createUser({email, hash: password}).then(response => {
          window.alert("User created!");
          this.setState({
            toProfileSettings: true,
            email: '',
            password: '',
            passwordConfirmation: ''
          });
      });
    }

  }

  render () {
    if (this.state.toProfileSettings === true) {
      return <Redirect to="/ProfileSettings" />
    }

    const {email, password, passwordConfirmation} = this.state;

    return(
      <div className="mx-2">
          <InputGroup className="mb-3">
            <FormControl
              type="email"
              placeholder="Email"
              aria-label="Email"
              aria-describedby="email-input"
              value={email}
              onChange={this.handleChangeInputEmail}
            />
          </InputGroup>

          <InputGroup className="mb-3">
            <FormControl
              type="password"
              placeholder="Password"
              aria-label="Password"
              aria-describedby="password-input"
              value={password}
              onChange={this.handleChangeInputPassword}
            />
          </InputGroup>

          <InputGroup className="mb-3">
            <FormControl
              type="password"
              placeholder="Confirm Password"
              aria-label="Confirm Password"
              aria-describedby="confirmation-input"
              value={passwordConfirmation}
              onChange={this.handleChangeInputPasswordConfirmation}
            />
          </InputGroup>

            <Button variant="primary" onClick={this.handleCreateUser}>
              Submit
            </Button>{' '}
        </div>
    );

  }
}

export default DropDownForm;
