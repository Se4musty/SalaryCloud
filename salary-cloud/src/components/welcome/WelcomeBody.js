import React from 'react';
import Dropdown from 'react-bootstrap/Dropdown';

import DropDownForm from './DropDownForm';

function WelcomeBody(props) {
  console.log(props);
  return (
    <div className="WelcomeBody container-fluid h-100" align="center">
      <h1>Welcome to SalaryCloud</h1>

      <Dropdown>
        <Dropdown.Toggle variant="success" id="dropdown-basic">
          Create Account
        </Dropdown.Toggle>

        <Dropdown.Menu className="text-center">
          <DropDownForm handleLogIn={props.handleLogIn} />
        </Dropdown.Menu>
      </Dropdown>
    </div>
  );
}

export default WelcomeBody;
