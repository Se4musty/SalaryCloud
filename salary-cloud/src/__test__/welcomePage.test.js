import React from 'react';
import WelcomeBody from '../components/welcome/WelcomeBody';
import DropDownForm from '../components/welcome/DropDownForm';
import Dropdown from 'react-bootstrap/Dropdown';
import renderer from 'react-test-renderer';
import {render, fireEvent, screen} from '@testing-library/react'

import Adapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme'
import {shallow} from 'enzyme';

Enzyme.configure({ adapter: new Adapter() })

describe ('Welcome Page', () => {
  it ('should render the usual html without crashing', () => {
    const body = renderer.create(<WelcomeBody />).toJSON();
    expect(body).toMatchSnapshot();
  });

  describe ('Create Account DropDownForm on Welcome Page', () => {
    it ('should render the usual html without crashing', () => {
      const dropdown = renderer.create(
        <DropDownForm loggedIn={false} handleLogIn={() => {return true}}/>
      ).toJSON();
      expect(dropdown).toMatchSnapshot();
    });

    it ('should render input groups that update onChange, but retain input', () => {
      // const dropdown = shallow(<DropDownForm loggedIn={false} handleLogIn={() => {return true}}/>);
      // var handleInputChange = (input) => { preventDefault: () => console.log('preventDefault')};
      // console.log(dropdown.debug());
      // const inputGroup = body.find('.input-group');
      // expect(inputGroup).toHaveLength(3);
      // expect(inputGroup[0].find('input').prop('type')).toEqual('email');
      // expect(inputGroup[1].find('input').prop('type')).toEqual('password');
      // expect(inputGroup[2].find('input').prop('type')).toEqual('password');
      // // Check for Submit Button
      // const buttons = body.find('button');
      // expect(buttons).to.have.lengthOf(2);
      // expect(buttons[1].text()).toEqual('Submit');
    });

    it ("should remove the dropdown on click", () => {
      render(<WelcomeBody />);
      expect(screen.queryByText('Create Account')).toBeInTheDocument();
    });

  });

});
