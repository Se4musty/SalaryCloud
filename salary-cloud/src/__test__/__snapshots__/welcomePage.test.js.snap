// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome Page Create Account DropDownForm on Welcome Page should render the usual html without crashing 1`] = `
<div
  className="mx-2"
>
  <div
    className="mb-3 input-group"
  >
    <input
      aria-describedby="email-input"
      aria-label="Email"
      className="form-control"
      data-testid="email-input"
      onChange={[Function]}
      placeholder="Email"
      type="email"
      value=""
    />
  </div>
  <div
    className="mb-3 input-group"
  >
    <input
      aria-describedby="password-input"
      aria-label="Password"
      className="form-control"
      data-testid="password-input"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
  </div>
  <div
    className="mb-3 input-group"
  >
    <input
      aria-describedby="confirmation-input"
      aria-label="Confirm Password"
      className="form-control"
      data-testid="confirm-password-input"
      onChange={[Function]}
      placeholder="Confirm Password"
      type="password"
      value=""
    />
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Submit
  </button>
</div>
`;

exports[`Welcome Page Create Account Dropdown on Welcome Page should render the usual contents without crashing 1`] = `
<div
  className="mx-2"
>
  <div
    className="mb-3 input-group"
  >
    <input
      aria-describedby="email-input"
      aria-label="Email"
      className="form-control"
      data-testid="email-input"
      onChange={[Function]}
      placeholder="Email"
      type="email"
      value=""
    />
  </div>
  <div
    className="mb-3 input-group"
  >
    <input
      aria-describedby="password-input"
      aria-label="Password"
      className="form-control"
      data-testid="password-input"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
  </div>
  <div
    className="mb-3 input-group"
  >
    <input
      aria-describedby="confirmation-input"
      aria-label="Confirm Password"
      className="form-control"
      data-testid="confirm-password-input"
      onChange={[Function]}
      placeholder="Confirm Password"
      type="password"
      value=""
    />
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Submit
  </button>
</div>
`;

exports[`Welcome Page Create Account Dropdown on Welcome Page should render the usual html without crashing 1`] = `
<div
  className="mx-2"
>
  <div
    className="mb-3 input-group"
  >
    <input
      aria-describedby="email-input"
      aria-label="Email"
      className="form-control"
      data-testid="email-input"
      onChange={[Function]}
      placeholder="Email"
      type="email"
      value=""
    />
  </div>
  <div
    className="mb-3 input-group"
  >
    <input
      aria-describedby="password-input"
      aria-label="Password"
      className="form-control"
      data-testid="password-input"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
  </div>
  <div
    className="mb-3 input-group"
  >
    <input
      aria-describedby="confirmation-input"
      aria-label="Confirm Password"
      className="form-control"
      data-testid="confirm-password-input"
      onChange={[Function]}
      placeholder="Confirm Password"
      type="password"
      value=""
    />
  </div>
  <button
    className="btn btn-primary"
    disabled={false}
    onClick={[Function]}
    type="button"
  >
    Submit
  </button>
</div>
`;

exports[`Welcome Page should render the usual contents without crashing 1`] = `
<div
  align="center"
  className="WelcomeBody container-fluid h-100"
>
  <h1>
    Welcome to SalaryCloud
  </h1>
  <div
    className="dropdown"
    onKeyDown={[Function]}
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-success"
      disabled={false}
      id="dropdown-basic"
      onClick={[Function]}
      type="button"
    >
      Create Account
    </button>
  </div>
</div>
`;

exports[`Welcome Page should render the usual html without crashing 1`] = `
<div
  align="center"
  className="WelcomeBody container-fluid h-100"
>
  <h1>
    Welcome to SalaryCloud
  </h1>
  <div
    className="dropdown"
    onKeyDown={[Function]}
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-success"
      disabled={false}
      id="dropdown-basic"
      onClick={[Function]}
      type="button"
    >
      Create Account
    </button>
  </div>
</div>
`;

exports[`should render correctly 1`] = `
<div
  align="center"
  className="WelcomeBody container-fluid h-100"
>
  <h1>
    Welcome to SalaryCloud
  </h1>
  <div
    className="dropdown"
    onKeyDown={[Function]}
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-success"
      disabled={false}
      id="dropdown-basic"
      onClick={[Function]}
      type="button"
    >
      Create Account
    </button>
  </div>
</div>
`;
